### PR TITLE
Fixes  open-vm-tools build and runtime errors

### PR DESCRIPTION
**Pull request by osmano807**
###### Things done
- [X] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [X] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Fixes issue #14436

### CODE REVIEW

File: nixos/modules/virtualisation/vmware-guest.nix [hunk 1]
```diff
@@ -25,6 +25,8 @@ in
         serviceConfig.ExecStart = "${open-vm-tools}/bin/vmtoolsd";
       };
 
+    environment.etc."vmware-tools".source = "${pkgs.open-vm-tools}/etc/vmware-tools/*";
```
Comment (jagajaga) [inline nixos/modules/virtualisation/vmware-guest.nix]: We mostly don't like such things. Why do you need that and can't it be avoided with patching some sources?

Comment (osmano807) [inline nixos/modules/virtualisation/vmware-guest.nix]: This commit fixes an error message on the host VMware

> The VMware Tools resume script did not run successfully in this virtual machine

> The VMware Tools power-on script did not run successfully in this virtual machine. If you have configured a custom power-on script in this virtual machine, make sure that it contains no errors. You can also submit a support request to report this issue.

I'm not sure if those scripts are really necessary, but getting an warning at startup and suspend/resume was annoying.

File: pkgs/applications/virtualization/open-vm-tools/default.nix [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
-  minorVersion = "0";
-  patchSet = "2476743";
-  version = "${majorVersion}.${minorVersion}-${patchSet}";
+  majorVersion = "10.0";
+  minorVersion = "7";
+  version = "${majorVersion}.${minorVersion}";
 
 in stdenv.mkDerivation {
   name = "open-vm-tools-${version}";
   src = fetchurl {
```
Comment (globin) [inline pkgs/applications/virtualization/open-vm-tools/default.nix]: This should use `fetchFromGitHub`

**Comment by zimbatm** [comment]
Fixed the fetchFromGitHub source and merged as 3ade1e7d3edea406ed4e2e4c692461cd8f685629

**Pull request closed without merging**